components/locid/src/parser/langid.rs

```diff
@@ -155,17 +162,25 @@ pub const fn parse_language_identifier_without_variants_from_iter(
             if let Ok(s) = subtags::Region::from_bytes_manual_slice(t, start, end) {
                 region = Some(s);
                 position = ParserPosition::Variant;
-            } else if subtags::Variant::from_bytes_manual_slice(t, start, end).is_ok() {
-                // We cannot handle variants in a const context
-                return Err(ParserError::InvalidSubtag);
+            } else if let Ok(v) = subtags::Variant::from_bytes_manual_slice(t, start, end) {
+                if variant.is_some() {
+                    // We cannot handle multiple variants in a const context
+                    return Err(ParserError::InvalidSubtag);
+                }
+                variant = Some(v);
+                position = ParserPosition::Variant;
             } else if matches!(mode, ParserMode::Partial) {
                 break;
             } else {
                 return Err(ParserError::InvalidSubtag);
             }
-        } else if subtags::Variant::from_bytes_manual_slice(t, start, end).is_ok() {
-            // We cannot handle variants in a const context
-            return Err(ParserError::InvalidSubtag);
+        } else if let Ok(v) = subtags::Variant::from_bytes_manual_slice(t, start, end) {
+            if variant.is_some() {
+                // We cannot handle multiple variants in a const context
+                return Err(ParserError::InvalidSubtag);
+            }
+            variant = Some(v);
+            position = ParserPosition::Variant;
```

Comment:
Nit: The `position` should already be `Variant`. You can add another debug assert:

```rust
debug_assert_eq!(position, ParserPosition::Variant);
```

I suggest putting that as the first line after the `else if`.

Reply:
Similarly, this causes compile failure because ParserPosition does not implement Debug (And if I add derived Debug, it wouldn't be const). Should I remove the position setting line instead?

Reply:
oh interesting, I guess `debug_assert_eq!` doesn't work in `const`.

One thing that _should_ work in `const` is

```rust
debug_assert!(matches!(position, ParserPosition::Variant))
```

Reply:
Done